build.sh

```diff
@@ -83,27 +90,51 @@ chmod a+x usr/sbin/policy-rc.d
 mount -o bind /proc /mnt/proc
 mount -o bind /dev /mnt/dev
 
-echo "deb http://ppa.launchpad.net/avsm/ocaml41+opam12/ubuntu trusty main" > /mnt/etc/apt/sources.list.d/ppa-opam.list
+echo "deb http://ppa.launchpad.net/avsm/ocaml41+opam12/ubuntu ${DISTROVER} main" > /mnt/etc/apt/sources.list.d/ppa-opam.list
 chown root /mnt/etc/apt/sources.list.d/ppa-opam.list
 
-echo "deb http://xenbits.xenproject.org/djs/linaro-xapi-4-4-talex5 ./" > /mnt/etc/apt/sources.list.d/linaro-xapi-4-4-talex5.list
-chown root /mnt/etc/apt/sources.list.d/linaro-xapi-4-4-talex5.list
+if ${INSTALL_XAPI} ; then
+  echo "deb http://xenbits.xenproject.org/djs/linaro-xapi-4-4-talex5 ./" > /mnt/etc/apt/sources.list.d/linaro-xapi-4-4-talex5.list
+  chown root /mnt/etc/apt/sources.list.d/linaro-xapi-4-4-talex5.list
+fi
 
-chroot /mnt apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 5B2D0C5561707B09
+# Add an optional apt-cacher-ng proxy
+if [ -n "${http_proxy}" ] ; then
+    echo "Acquire::http::Proxy \"${http_proxy}\";" > /mnt/etc/apt/apt.conf.d/proxy
```

Comment:
Looks like the image inherits the build system's proxy setup here? Could be a problem if the image is made to be used in a different network (or proxy is local).